src/logger/winston-logger.js

```diff
@@ -1,11 +1,38 @@
+import globalEnv from '../global-env'
 import winston from 'winston'
 import { LoggingWinston, express } from '@google-cloud/logging-winston'
 
+/**
+ *  @param {string} releaseBranch - 'master', 'preview', 'staging' or 'release'
+ *  @return {string} kubernetes namespace
+ */
+function getK8sNamespaceName(releaseBranch) {
+  switch(releaseBranch) {
+    case 'release':
+      return 'production'
+    default:
+      return releaseBranch
+  }
+}
+
 /**
  *  @return {Object} Winston logger instance
  */
 export function createProdLogger() {
   const loggingWinston = new LoggingWinston({
+    // For (new) stackdriver logging.
+    // If we don't specify resource object,
+    // default `resource.type` would be 'container';
+    // hence, logs will only be viewed in legacy stackdriver logging.
+    // See more: https://cloud.google.com/monitoring/kubernetes-engine/migration?hl=zh-tw#resource_type_changes
+    resource: {
+      type: 'k8s_container',
+      labels: {
+        cluster_name: 'twreporter',
+        namespace_name: getK8sNamespaceName(globalEnv.releaseBranch),
+        container_name: 'twreporter-website-v2'
+      }
+    },
     // for stackdriver error reporting
     serviceContext: {
       service: 'twreporter-react-server',
```

Comment:
Why does the serviceContext.version not match with the service version?

Reply:
Which service version do you mean? The package.json#version?

Reply:
Yes

Reply:
I will remove `serviceContext.version` here.

Reply:
I'm not sure if the entry removed will break the serviceContext object or not. I'm not objecting to specify the version but to make the meaning of the version clear.
https://cloud.google.com/error-reporting/reference/rest/v1beta1/ServiceContext

Reply:
it won't break  `serverContext` object since `serviceContext.version` is optional (https://github.com/googleapis/nodejs-logging-winston/blob/master/src/types/core.d.ts#L127).

Reply:
The interface declaration stands for the object for stackdriver logging. The service context entry is not a must. But for errorReporting event, the serviceContext does.

Reply:
I believe it won't break. In other setup, only `service: gke_instance`  is specified

Reply:
Just test the error reporting without `serviceContext.version`, and it reports error.
However, I plan to pass `globalEnv.pkgVersion` and use it as `serviceContext.version`.